core/spec/models/spree/payment_method/check_spec.rb

```diff
@@ -0,0 +1,77 @@
+require 'spec_helper'
+
+describe Spree::PaymentMethod::Check do
+  let(:order) { create(:order) }
+
+  context "#can_capture?" do
+    let(:payment) { create(:payment, order: order, state: state) }
+
+    context "with payment in state checkout" do
+      let(:state) { "checkout" }
+
+      it "returns true" do
+        expect(Spree::PaymentMethod::Check.new.can_capture?(payment)).to be_truthy
+      end
+    end
+
+    context "with payment in state pending" do
+      let(:state) { "pending" }
+
+      it "returns true" do
+        expect(Spree::PaymentMethod::Check.new.can_capture?(payment)).to be_truthy
+      end
+    end
+
+    context "with payment in state failed" do
+      let(:state) { "failed" }
+
+      it "returns false" do
+        expect(Spree::PaymentMethod::Check.new.can_capture?(payment)).to be_falsy
+      end
+    end
+  end
+
+  context "#can_void?" do
+    let(:payment) { create(:payment, order: order, state: state) }
+
+    context "with payment in state checkout" do
+      let(:state) { "checkout" }
+
+      it "returns true" do
+        expect(Spree::PaymentMethod::Check.new.can_void?(payment)).to be_truthy
+      end
+    end
+
+    context "with payment in state void" do
+      let(:state) { "void" }
+
+      it "returns false" do
+        expect(Spree::PaymentMethod::Check.new.can_void?(payment)).to be_falsy
+      end
+    end
+  end
+
+  context "#capture" do
+    it "succeds" do
+      expect(Spree::PaymentMethod::Check.new.capture).to be_success
+    end
+  end
+
+  context "#cancel" do
+    it "returns nil" do
+      expect(Spree::PaymentMethod::Check.new.cancel).to be_nil
+    end
+  end
+
+  context "#void" do
+    it "succeds" do
+      expect(Spree::PaymentMethod::Check.new.void).to be_success
+    end
+  end
+
+  context "#credit" do
+    it "succeds" do
+      expect(Spree::PaymentMethod::Check.new.credit).to be_success
```

Comment:
`Spree::PaymentMethod::Check` - can be referenced using `described_class`, and the `new` form of it is usually referenced using `subject`, so at the top you can do:

```ruby
subject { described_class.new }
# ...
expect(subject.credit).to ....
```